himl/secret_resolvers.py

```diff
@@ -29,17 +36,33 @@ def supports(self, secret_type):
     def resolve(self, secret_type, secret_params):
         aws_profile = secret_params.get("aws_profile", self.default_aws_profile)
         if not aws_profile:
-            raise Exception("Could not find the aws_profile in the secret params: {}".format(secret_params))
+            raise Exception("Could not find the aws_profile in the secret params for SSM secret: {}".format(secret_params))
 
         path = self.get_param_or_exception("path", secret_params)
         region_name = secret_params.get("region_name", "us-east-1")
         ssm = SimpleSSM(aws_profile, region_name)
         return ssm.get(path)
 
-    def get_param_or_exception(self, key, params):
-        if key not in params:
-            raise Exception("Could not find required key '{}' in the secret params: {}".format(key, params))
-        return params[key]
+
+class S3SecretResolver(SecretResolver):
```

Comment:
S3 may not be the right "secret store".
Why we need S3 to store secrets and SSM can't do ?

Reply:
SSM has a character limit. We're planning to store some kube configs (eg. AKS k8s config), so that we don't keep them in git. Vault was another solution taken into consideration, but it's quite cumbersome to setup (tokens and all that, plus the well known availability issues). Hence, fetching these from S3 by using a simple client based on the boto profile was the most straightforward solution.